src/main/java/edu/illinois/cs/dt/tools/fixer/CleanerFixerPlugin.java

```diff
@@ -0,0 +1,211 @@
+        final List<String> elements = new ArrayList<>(project.getCompileClasspathElements());
+        elements.addAll(project.getRuntimeClasspathElements());
+        elements.addAll(project.getTestClasspathElements());
+
+        return String.join(File.pathSeparator, elements);
+    }
+
+    @Override
+    public void execute(final MavenProject project) {
+        this.project = project;
+
+        final Option<Runner> runnerOption = RunnerFactory.from(project);
+        final ErrorLogger logger = new ErrorLogger(project);
+
+        System.out.println("DIAGNOSER_MODULE_COORDINATES: " + logger.coordinates());
+
+        logger.runAndLogError(() -> {
+            logger.writeSubjectProperties();
+            this.classpath = classpath();
+
+            if (runnerOption.isDefined()) {
+                this.runner = InstrumentingSmartRunner.fromRunner(runnerOption.get());
+
+                if (!Files.exists(DetectorPathManager.originalOrderPath())) {
+                    Files.write(DetectorPathManager.originalOrderPath(), DetectorPlugin.getOriginalOrder(project));
+                }
+
+                // TODO: Modify so that it will read parameters directly from the
+                setupAndApplyFix();
+            } else {
+                final String errorMsg = "Module is not using a supported test framework (probably not JUnit).";
+                TestPluginPlugin.info(errorMsg);
+                logger.writeError(errorMsg);
+            }
+
+            return null;
+        });
+    }
+
+    private void setupAndApplyFix() throws Exception {
+        // Get all test source files
+        final List<Path> testFiles = new ArrayList<>();
+        try (final Stream<Path> paths = Files.walk(Paths.get(project.getBuild().getTestSourceDirectory()))) {
+            paths.filter(Files::isRegularFile)
+                    .forEach(testFiles::add);
+        }
+
+        // TODO: Remove these and read from minimized files
+        final String polluterTestName = System.getProperty("dt_fixer.polluter");
+        final String cleanerTestName = System.getProperty("dt_fixer.cleaner");
+        final String victimTestName = System.getProperty("dt_fixer.victim");
+
+        final Optional<JavaMethod> cleanerMethodOpt = JavaMethod.find(cleanerTestName, testFiles, classpath);
+        final Optional<JavaMethod> victimMethodOpt = JavaMethod.find(victimTestName, testFiles, classpath);
+
+        if (!cleanerMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + cleanerTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        if (!victimMethodOpt.isPresent()) {
+            TestPluginPlugin.error("Could not find method " + victimTestName);
+            TestPluginPlugin.error("Tried looking in: " + testFiles);
+            return;
+        }
+
+        applyFix(polluterTestName, cleanerMethodOpt.get(), victimMethodOpt.get());
```

Comment:
Does not have to be with this pull request but we should have applyFix take in a location for where to output the Java file that contains the "fixed" code or an option to directly replace the existing test source file. The former is for usability while the latter should speed up our experiments so that we can quickly just review the changes of ASDF before committing and making pull requests.